model.py

```diff
@@ -125,29 +132,27 @@ def Train(self,
     # Number of ensembles in the model
     nb_ensembles = int(np.ceil(self.nb_trees / self.nb_trees_per_ensemble))
 
-    prev_rmse = np.inf
+    prev_score = np.inf
 
     # Train all trees
     for tree_index in range(self.nb_trees):
-      if tree_index == 0:
-        # First tree, start with initial scores (mean of labels)
-        gradients = self.ComputeGradientForLossFunction(
-            y, self.init_score[:len(y)])
-      else:
-        # Update gradients of all training instances on loss l
-        if update_gradients:
-          gradients = self.ComputeGradientForLossFunction(
-              y_ensemble, self.Predict(X_ensemble))  # type: ignore
+      # Compute sensitivity
+      delta_g = 3 * np.square(self.l2_threshold)
+      delta_v = min(self.l2_threshold / (1 + self.l2_lambda),
+                    2 * self.l2_threshold * math.pow(
+                      (1 - self.learning_rate), tree_index))
+
+      # Privacy budget allocated to each tree
+      tree_privacy_budget = np.divide(self.privacy_budget, nb_ensembles * self.loss_.K)
+
       current_tree_for_ensemble = tree_index % self.nb_trees_per_ensemble
       if current_tree_for_ensemble == 0:
         # Initialize the dataset and the gradients
         X_ensemble = np.copy(X)
         y_ensemble = np.copy(y)
-        prev_rmse = np.inf
+        prev_score = np.inf
         update_gradients = True
-        if tree_index > 0:
-          gradients = self.ComputeGradientForLossFunction(
-              y_ensemble, self.Predict(X_ensemble))
+        # gradient initialization will happen later in the per-class-loop
```

Comment:
Regression output remains the same as before. :+1:

Reply:
This `if` statement only is reached when there are multiple ensembles (not the case in `example.py`). Can you check if with e.g. 100 trees and 20 trees per ensemble there's no impact?
Also, irrespective of the output, why shouldn't we reinitialise the gradients here? Since these trees are independent from the previous ones and work on different data I'd expect the need of initialising the gradients again (note that the choice of initialisation is another topic, I'm not sure which is best in this case as this is not mentioned in the original paper)